go/vt/sqlparser/sql.y

```diff
@@ -2040,6 +2041,30 @@ function_call_keyword:
   {
     $$ = &ConvertUsingExpr{Expr: $3, Type: $5}
   }
+| SUBSTR openb column_name ',' value_expression closeb
```

Comment:
I think this can be improved by using a new non-terminal `substr_keyword`, but I'll do it myself, because I'm changing the ast parts for the subquery support.